tools/apiview/parsers/rust-api-parser/src/utils/lineIdUtils.ts

```diff
@@ -3,6 +3,50 @@ import { ReviewLine } from "../models/apiview-models";
 
 export const lineIdMap = new Map<string, string>();
 
+/**
+ * Sanitizes a string for use in line IDs while preserving meaningful distinctions
+ * @param input The input string to sanitize
+ * @returns A sanitized string suitable for line IDs
+ */
+export function sanitizeForLineId(input: string): string {
+  return input
+    // Handle dereferencing and pointer operations first (preserve workflow context)
+    .replace(/\*mut\s+/g, "deref_mut_")
+    .replace(/\*const\s+/g, "deref_const_")
+    .replace(/\*\s*/g, "deref_")
+    // Replace reference symbols with meaningful text (preserve reference context)
+    .replace(/&mut\s+/g, "refmut_")
+    .replace(/&\s*/g, "ref_")
+    // Handle module path separators with context preservation
+    .replace(/::/g, "_path_")
+    // Handle trait bounds and where clauses
+    .replace(/\bwhere\b/g, "_where_")
+    .replace(/\+\s*/g, "_and_")
+    // Replace generic brackets with meaningful separators
+    .replace(/</g, "_of_")
+    .replace(/>/g, "_end_")
+    // Handle function pointers and closures with context
+    .replace(/\bfn\s*\(/g, "fnptr_")
+    .replace(/\|([^|]*)\|/g, "_closure_$1_")
+    // Replace common punctuation with meaningful separators
+    .replace(/\s+/g, "_") // spaces to underscores
+    .replace(/:/g, "_")   // colons to underscores
+    .replace(/,/g, "_")   // commas to underscores
+    .replace(/\(/g, "_")  // parentheses
+    .replace(/\)/g, "_")
+    .replace(/\[/g, "_")  // brackets
+    .replace(/\]/g, "_")
+    .replace(/\{/g, "_")  // braces
+    .replace(/\}/g, "_")
+    .replace(/'/g, "_lifetime_") // lifetime parameters
+    .replace(/\$/g, "_dollar_")  // macro variables
```

Comment:
Speaking of macros, do attribute macros get a line ID? If so, you'll also need to handle `!` and `#`. You already have `[` and `]`, but I would use names like `_exclamation_` and `_hash_` or something.